observationtree/src/main/scala/explore/observationtree/ObsList.scala

```diff
@@ -44,48 +53,118 @@ object ObsList {
 
   implicit val propsReuse: Reusability[Props] = Reusability.derive
 
-  def createObservation[F[_]: Applicative](
-    obsId:      Observation.Id,
-    name:       String
-  )(implicit c: TransactionalClient[F, ObservationDB]): F[Unit] =
-    ObsQueries.ProgramCreateObservations
-      .execute[F](
-        CreateObservationInput(programId = "p-2", observationId = obsId.assign, name = name.assign)
-      )
-      .void
+  val obsListMod = new KIListMod[IO, ObsSummary, Observation.Id](ObsSummary.id)
 
-  def deleteObservation[F[_]: Applicative](id: Observation.Id)(implicit
-    c:                                         TransactionalClient[F, ObservationDB]
-  ): F[Unit] =
-    ObsQueries.ProgramDeleteObservation.execute[F](id).void
+  class Backend($ : BackendScope[Props, Unit]) {
 
-  protected val component =
-    ScalaComponent
-      .builder[Props]
-      .render_P { props =>
-        def deleteLocal(id: Observation.Id, focus: Option[ObsSummary]) =
-          props.observations.mod(l => l.filterNot(_.id === id)) *>
-            props.focused.set(focus.map(s => FocusedObs(s.id)))
-
-        def createLocal(name: String): IO[Observation.Id] =
-          IO(Observation.Id(PosLong.unsafeFrom(Random.nextInt().abs.toLong + 1))).flatTap { oid =>
-            props.observations.mod(l =>
-              l :+ ObsSummary(oid, name.some, observationTarget = none)
-            ) *>
-              props.focused.set(FocusedObs(oid).some)
+    def createObservation[F[_]](
+      obsId: Observation.Id,
+      name:  Option[String]
+    )(implicit
+      F:     ApplicativeError[F, Throwable],
+      c:     TransactionalClient[F, ObservationDB]
+    ): F[Unit] =
+      ObsQueries.ProgramCreateObservation
+        .execute[F](
+          CreateObservationInput(programId = "p-2",
+                                 observationId = obsId.assign,
+                                 name = name.orIgnore
+          )
+        )
+        .void
+        .handleErrorWith { _ =>
+          ProgramUndeleteObservation.execute(obsId).void
+        }
+
+    def deleteObservation[F[_]: Applicative](id: Observation.Id)(implicit
+      c:                                         TransactionalClient[F, ObservationDB]
+    ): F[Unit] =
+      ObsQueries.ProgramDeleteObservation.execute[F](id).void
+
+    private def setObsWithIndex(
+      observations:       View[ObservationList],
+      focused:            View[Option[Focused]],
+      obsId:              Observation.Id,
+      obsWithIndexSetter: Adjuster[ObservationList, obsListMod.ElemWithIndex],
+      nextToFoucs:        Option[ObsSummary]
```

Comment:
Typo. Unless "Foucs" is a thing is Spanish.

Reply:
Good catch, thank you